graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/InputArgumentResolver.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import com.netflix.graphql.dgs.InputArgument
+import com.netflix.graphql.dgs.exceptions.DgsInvalidInputArgumentException
+import graphql.schema.DataFetchingEnvironment
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.core.KotlinDetector
+import org.springframework.core.MethodParameter
+import org.springframework.core.annotation.MergedAnnotation
+import org.springframework.core.convert.TypeDescriptor
+import org.springframework.core.convert.converter.ConditionalGenericConverter
+import org.springframework.core.convert.converter.GenericConverter
+import org.springframework.core.convert.support.DefaultConversionService
+import java.util.Optional
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+class InputArgumentResolver(
+    inputObjectMapper: InputObjectMapper
+) : ArgumentResolver {
+
+    companion object {
+        private val logger: Logger = LoggerFactory.getLogger(InputArgumentResolver::class.java)
+    }
+
+    private val argumentNameCache: ConcurrentMap<MethodParameter, String> = ConcurrentHashMap()
+    private val conversionService = DefaultConversionService()
+
+    init {
+        conversionService.addConverter(ObjectMapperConverter(inputObjectMapper))
+    }
+
+    private class ObjectMapperConverter(private val inputObjectMapper: InputObjectMapper) : ConditionalGenericConverter {
```

Comment:
[Here](https://github.com/spring-projects/spring-framework/blob/main/spring-core/src/main/java/org/springframework/core/convert/support/DefaultConversionService.java#L87) are the default converters we get from Spring's ConversionService. The collection and scalar converters take care of a lot of what we were doing before, but there's nothing out of the box to handle conversion of a `Map<String, ?>` to an Object, which is basically what `InputObjectMapper` is doing. That said, I plan to simplify that in a future PR as well, because I think we can leverage `DataBinder` / `PropertyEditor` constructs from Spring to handle what it does now.